src/figuresClasses.ts

```diff
@@ -1,19 +1,112 @@
+
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle'| 'circle' |'rectangle';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea():number
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  color: Color;
+
+  a:number;
+
+  b:number;
+
+  c:number;
+
```

Comment:
let's use the next syntax in constructor for creating new properties instead of repeating the same code 2 times (declaring props here, and added them in constructor): `public color: Color`
just add input values in such way, and you don't need to declare it and assign a value 2 times